### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by samuel-oci**
### Description
Initial commit for the OCI object storage repository plugin.
 
### Issues Resolved
Initial implementation for OCI object storage repository plugin.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,158 @@
...
 
-TODO: Fill this README out!
+## Build and install the plugin
+To build the plugin zip distribution file
+```bash
+mvn clean install -PAdvanced
+```
+Install the plugin on your Elasticsearch cluster
+```bash
+OS_HOME=<YOUR OPENSEARCH INSTALLATION PATH HERE> # (e.g. /Users/saherman/opensearch)
```
Comment (dblock) [inline README.md:10]: Recommend expanding `OS` into `OPENSEARCH` because OS is too operating system-y.

File: oci-repository-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
+# Build and install the plugin
```
Comment (dblock) [inline oci-repository-plugin/README.md:1]: This duplicates parent README?

**Review by dblock — COMMENTED** [review]
Glad to see this coming in! I did a cursory review. 

1. Looks like you have some duplication and leftover Elasticsearch in READMEs. 
2. I recommend adding CI in this first PR and making sure the source code builds and runs tests from the start.
3. Consider explaining what this project is in README, and moving some of the existing README content into DEVELOPER_GUIDE.

File: oci-repository-plugin/README.md (hunk 2 above)
Comment (samuel-oci) [reply oci-repository-plugin/README.md]: Looks like it slipped through the cracks, removed the duplicated README with the old references to ES.

**Comment by samuel-oci** [comment]
Thank you for the review @dblock !

> Glad to see this coming in! I did a cursory review.
> 
>     1. Looks like you have some duplication and leftover Elasticsearch in READMEs.
removed
> 
>     2. I recommend adding CI in this first PR and making sure the source code builds and runs tests from the start.
That would be a great idea! I would have to wait until I get the admin permissions on the repo. Looks like github doesn't let me setup any actions at the moment.
> 
>     3. Consider explaining what this project is in README, and moving some of the existing README content into DEVELOPER_GUIDE.
Added explanation about the plugin in README, I also like the idea of separating it to DEVELOPER_GUIDE.

**Comment by dblock** [comment]
If you PR a new CI GHA I think I can click a button to kick it off for you while you wait for access.

**Comment by samuel-oci** [comment]
> If you PR a new CI GHA I think I can click a button to kick it off for you while you wait for access.

@dblock added a CI looks like it's all passing.

**Comment by samuel-oci** [comment]
> > If you PR a new CI GHA I think I can click a button to kick it off for you while you wait for access.
> 
> @dblock added a CI looks like it's all passing.

Hi @dblock let me know if you feel it's good to merge at this point. The CI check also runs all the tests which seems to be passing.

**Pull request merged**